src/App/App.js

```diff
@@ -43,28 +40,36 @@ export default class App extends React.Component {
           return response;
         }
       })
-      .then(response => this.setState({sightings: [response, ...this.state.sightings], redirect: true, error: ''}))
-      .catch(error => this.setState({error: error}));
+      .then(data => {
+        updateSightings([data, ...sightings]);
+        updateError('');
+        updateRedirect(true);
+      })
+      .catch(error => updateError(error))
   }
 
-  render() {
-    return (
-      <>
-        <h1>👁 SkyWatcher 👁</h1>
-        <nav>
-          <NavLink exact to='/'>Sightings</NavLink>
-          <NavLink to='/report'>Report a new sighting</NavLink>
-        </nav>
+  return (
+    <>
+      <h1>👁 SkyWatcher 👁</h1>
+      <nav>
+        <NavLink to='/'>Sightings</NavLink>
+        <NavLink to='/report' onClick={() => updateRedirect(false)}>Report a new sighting</NavLink>
+      </nav>
 
-        {this.state.error && <p className="error">{this.state.error.message}</p>}
-        {!this.state.sightings.length && <p>Loading ...</p>}
-        {this.state.sightings.length && !this.state.error &&
-          <Route exact path="/"
-            render={() => <Sightings sightings={this.state.sightings} />} />
-        }
-        <Route exact path='/report' render={() => <Form addNewSighting={this.addNewSighting} />} />
-        {this.state.redirect && <Redirect exact to="/" />}
-      </>
-    )
-  }
-}
\ No newline at end of file
+      {error && <p className='error'>{error.message}</p>}
+
+      {!sightings.length && <p>Loading ...</p>}
+
+      {sightings.length && !error &&
+        <Routes>
+          <Route exact path='/' element={<Sightings sightings={sightings} />} />
+          <Route exact path='/report' element={<Form addNewSighting={addNewSighting} />} />
+        </Routes>
+      }
+
+      {redirect && <Navigate replace to='/' />}
```

Comment:
I haven't coded in React IRL since React-Router V6 came out and I haven't been in M3 for over a year, so I don't have full context for these exercises, so please don't take my notes as gospel.

This totally works - but it's a lot more code than just using the `useNavigate` hook. Maybe it's important not to use that hook so as to not introduce more hooks to the students? 

I think I'd probably vote for the `useNavigate` hook, otherwise. Here is my reasoning: 
- no need to track state to know if the app should redirect or not.
- no extra line at the bottom of the return.
- the docs suggest that the `<Navigate>` component exists for use in class components, since they can't utilize the `useNavigate` hook. 

Ultimately, <Navigate> is using `useNavigate` under the hood anyways - it's just a matter of how much code you want to have, unless there are other considerations I'm missing.

Reply:
thank you! I'll make that change!